src/Glpi/Controller/Security/MFAController.php

```diff
@@ -0,0 +1,155 @@
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller\Security;
+
+use Glpi\Controller\AbstractController;
+use Glpi\Exception\AuthenticationFailedException;
+use Glpi\Http\Firewall;
+use Glpi\Security\Attribute\SecurityStrategy;
+use Glpi\Security\TOTPManager;
+use Html;
+use Session;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+final class MFAController extends AbstractController
+{
+    #[Route(
+        path: "/MFA/Setup",
+        name: "mfa_setup",
+    )]
+    #[SecurityStrategy(Firewall::STRATEGY_NO_CHECK)]
+    public function setup(Request $request): Response
+    {
+        global $CFG_GLPI;
+        if (!isset($_SESSION['mfa_pre_auth'])) {
+            return new RedirectResponse($CFG_GLPI['root_doc'] . '/front/login.php');
+        }
+        return new StreamedResponse(static function () {
+            $totp = new TOTPManager();
+            $totp->showTOTPSetupForm((int) $_SESSION['mfa_pre_auth']['user']['id']);
+        });
+    }
+
+    #[Route(
+        path: "/MFA/Prompt",
+        name: "mfa_prompt",
+        methods: ['GET']
+    )]
+    #[SecurityStrategy(Firewall::STRATEGY_NO_CHECK)]
+    public function prompt(Request $request): Response
+    {
+        $totp = new TOTPManager();
+        return new StreamedResponse(static function () use ($totp) {
+            $totp->showTOTPPrompt();
+        });
+    }
+
+    #[Route(
+        path: "/MFA/Verify",
+        name: "mfa_verify_totp",
+        methods: ['POST']
+    )]
+    #[SecurityStrategy(Firewall::STRATEGY_NO_CHECK)]
+    public function verify(Request $request): Response
+    {
+        global $CFG_GLPI;
+
+        $from_login = isset($_SESSION['mfa_pre_auth']);
+        $users_id = $from_login ? (int) $_SESSION['mfa_pre_auth']['user']['id'] : (int) Session::getLoginUserID();
+        if (!$users_id) {
+            return new RedirectResponse($CFG_GLPI['root_doc'] . '/MFA/Prompt');
```

Comment:
In the symfony controllers, please use `$request->getBasePath()` instead of `$CFG_GLPI['root_doc']`.

```suggestion
            return new RedirectResponse($request->getBasePath() . '/MFA/Prompt');
```